core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java

```diff
@@ -459,6 +460,11 @@ public void begin() throws RepositoryException {
 		}
 	}
 
+	@Override
+	public void begin(TransactionSetting... settings) {
+		throw new UnsupportedOperationException("SPARQLConnection does not support transaction settings");
```

Comment:
I think silently ignoring (possibly logging a warning) might be better here. Not the most important thing right now though.

Reply:
I prefer an exception when calling code that doesn't do what the user expects.

Reply:
The default method does backwards compatibility by calling `begin()` by default. This method should also call `begin()` for backwards compatibility unless there is a specific reason for not following the previous transactional implementation which hasn't been deleted or modified here.

Reply:
that's actually a good point

Reply:
fixed